python/SNS_ENCRYPTED_TOPIC_CHECK/SNS_ENCRYPTED_TOPIC_CHECK_test.py

```diff
@@ -0,0 +1,254 @@
+                "KmsMasterKeyId": "arn:aws:kms:ap-southeast-1:123456789012:key/86a9f691-c02f-4046-9360-903afec68edc"
+            }}
+
+        SNS_CLIENT_MOCK.list_topics = MagicMock(return_value=list_topics_result)
+        SNS_CLIENT_MOCK.get_topic_attributes = MagicMock(return_value=get_topic_attributes_result)
+        rule_parameters = '{"KmsKeyId": "arn:aws:kms:ap-southeast-1:123456789012:key/86a9f691-c02f-4046-9360-903afec68edc"}'
+        lambda_result = RULE.lambda_handler(build_lambda_scheduled_event(rule_parameters), {})
+        expected_response = [build_expected_response(
+            'COMPLIANT',
+            'arn:aws:sns:ap-southeast-1:123456789012:testSNS'
+        )]
+        assert_successful_evaluation(self, lambda_result, expected_response, len(lambda_result))
+
+class NonCompliantResourcesTest(unittest.TestCase):
+
+    def test_scenario_3_non_compliant_resources_without_key(self):
+        list_topics_result = {"Topics": [{"TopicArn": "arn:aws:sns:ap-southeast-1:123456789012:dynamodbtopic"}]}
+
+        get_topic_attributes_result = {"Attributes": {}}
+
+        SNS_CLIENT_MOCK.list_topics = MagicMock(return_value=list_topics_result)
+        SNS_CLIENT_MOCK.get_topic_attributes = MagicMock(return_value=get_topic_attributes_result)
+        lambda_result = RULE.lambda_handler(build_lambda_scheduled_event({}), {})
+        expected_response = [build_expected_response(
+            'NON_COMPLIANT',
+            'arn:aws:sns:ap-southeast-1:123456789012:dynamodbtopic',
+            annotation="The Amazon Simple Notification Service topic is not encrypted."
+        )]
+        assert_successful_evaluation(self, lambda_result, expected_response, len(lambda_result))
+
+    def test_scenario_5_non_compliant_resources_with_key(self):
+        list_topics_result = {"Topics": [{"TopicArn": "arn:aws:sns:ap-southeast-1:123456789012:testSNS"}]}
+
+        get_topic_attributes_result = {
+            "Attributes": {
+                "KmsMasterKeyId": "arn:aws:kms:ap-southeast-1:123456789012:key/86a9f691-c02f-4046-9360-903afec68edc"
+            }}
+
+        SNS_CLIENT_MOCK.list_topics = MagicMock(return_value=list_topics_result)
+        SNS_CLIENT_MOCK.get_topic_attributes = MagicMock(return_value=get_topic_attributes_result)
+        rule_parameters = '{"KmsKeyId": "arn:aws:kms:ap-southeast-1:123456789012:key/99a9f661-c02f-4046-9360-9334dex68gdc"}'
+        lambda_result = RULE.lambda_handler(build_lambda_scheduled_event(rule_parameters), {})
+        expected_response = [build_expected_response(
+            'NON_COMPLIANT',
+            'arn:aws:sns:ap-southeast-1:123456789012:testSNS',
+            annotation="This SNS topic is not encrypted with KMS Key {KmsKeyId}: ['arn:aws:kms:ap-southeast-1:123456789012:key/99a9f661-c02f-4046-9360-9334dex68gdc']"
+        )]
+        assert_successful_evaluation(self, lambda_result, expected_response, len(lambda_result))
+
+
+class NotApplicable(unittest.TestCase):
```

Comment:
Can you reorder the classes by scenario 1 to X. It will be easier to read.

Reply:
Rearranged the classes